modules/RTC/RTC.js

```diff
@@ -260,94 +362,134 @@ export default class RTC extends Listenable {
+     * @param {VideoType|undefined} videoType
+     * @param {string} ssrc
+     * @param {boolean} muted
+     */
+    _createRemoteTrack (owner,
+                        stream, track, mediaType, videoType, ssrc, muted) {
+        const remoteTrack
+            = new JitsiRemoteTrack(
+                this, this.conference, owner, stream, track,
+                mediaType, videoType, ssrc, muted);
+        const remoteTracks
+            = this.remoteTracks[owner] || (this.remoteTracks[owner] = {});
+
         if (remoteTracks[mediaType]) {
-            logger.warn("Overwriting remote track!", resource, mediaType);
+            logger.error("Overwriting remote track!", owner, mediaType);
         }
         remoteTracks[mediaType] = remoteTrack;
-        return remoteTrack;
+
+        this.eventEmitter.emit(RTCEvents.REMOTE_TRACK_ADDED, remoteTrack);
     }
 
     /**
-     * Removes all JitsiRemoteTracks associated with given MUC nickname (resource
-     * part of the JID). Returns array of removed tracks.
+     * Removes all JitsiRemoteTracks associated with given MUC nickname
+     * (resource part of the JID). Returns array of removed tracks.
      *
-     * @param {string} resource - The resource part of the MUC JID.
+     * @param {string} owner - The resource part of the MUC JID.
      * @returns {JitsiRemoteTrack[]}
      */
-    removeRemoteTracks (resource) {
-        var removedTracks = [];
-        var removedAudioTrack = this.removeRemoteTrack(resource, MediaType.AUDIO);
-        var removedVideoTrack = this.removeRemoteTrack(resource, MediaType.VIDEO);
+    removeRemoteTracks (owner) {
+        const removedTracks = [];
 
-        removedAudioTrack && removedTracks.push(removedAudioTrack);
-        removedVideoTrack && removedTracks.push(removedVideoTrack);
+        if (this.remoteTracks[owner]) {
+            const removedAudioTrack
+                = this.remoteTracks[owner][MediaType.AUDIO];
+            const removedVideoTrack
+                = this.remoteTracks[owner][MediaType.VIDEO];
 
-        delete this.remoteTracks[resource];
+            removedAudioTrack && removedTracks.push(removedAudioTrack);
+            removedVideoTrack && removedTracks.push(removedVideoTrack);
 
+            delete this.remoteTracks[owner];
+        }
         return removedTracks;
     }
 
     /**
-     * Removes specified track type associated with given MUC nickname
-     * (resource part of the JID). Returns removed track if any.
+     * Finds remote track by it's stream and track ids.
+     * @param {string} streamId the media stream id as defined by the WebRTC
+     * @param {string} trackId the media track id as defined by the WebRTC
+     * @return {JitsiRemoteTrack|undefined}
+     * @private
+     */
+    _getRemoteTrackById (streamId, trackId) {
+        let result = undefined;
+
+        // .find will break the loop once the first match is found
+        Object.keys(this.remoteTracks).find(function(endpoint) {
```

Comment:
you can use an arrow function here and save passing `this` as the last parameter.